cpp/src/arrow/dataset/scanner.cc

```diff
@@ -74,6 +74,13 @@ class ScannerRecordBatchReader : public RecordBatchReader {
                                     TaggedRecordBatchIterator delegate)
       : schema_(std::move(schema)), delegate_(std::move(delegate)) {}
 
+  ~ScannerRecordBatchReader() {
+    auto st = this->Close();
+    if (!st.ok()) {
+      ARROW_LOG(WARNING) << "ScannerRecordBatchReader failed in finalzing reading.";
```

Comment:
Is there any need to override the destructor here? Also this error message is less informative than the one on the base class.

Reply:
Yes, may be now we don't need this as we modified the base class.